draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -52,34 +52,83 @@ TODO Abstract
+Assume that Service A needs to call Service B. For simplicity, this document focuses on REST services,
+and the service-to-service call consists of a single HTTP request and its response.
+We define the credentials that both services should possess and how they are used to protect the HTTP exchange.
+
+There are multiple deployment styles in use today, and they result in different security properties.
+We propose to address them differently.
+
+* Many use cases have various middleboxes inserted between pairs of services, resulting in a transport layer
+that is not end-to-end encrypted. We propose to address these use cases by protecting the REST messages at the application
+level ({{app-level}}).
+
+* The other commonly deployed architecture has a mutual-TLS connection between each pair of services. This setup
+can be addressed by a simpler solution ({{mutual-tls}}).
+
+It is an explicit goal of this protocol that a service deployment, and in fact a single call chain,
+can include both architectures. In other words, Service A can call Service B with mutual TLS protection,
+while the next call to Service C is protected at the application level.
+
+For application-level protection we currently propose two alternative solutions, one inspired by DPoP {{?RFC9449}} and
+one which is a profile of HTTP Message Signatures {{!RFC9421}}. The design team believes that we need to pick
+one of these two alternatives for standardization, once we have understood their pros and cons.
 
 ## Deployment Architecture and Message Flow
 
+Regardless of the transport between the workloads, we assume the following logical architecture:
+
 ~~~ aasvg
-+---------+
-|         |
-| This    |
-| is      |
-| a       |
-| box     |
-|         |
-+---------+
++------------+               +------------+
+|            |               |            |
+|            |               | Workload B |
+| Workload A |==============>|            |
+|            |               +------------+
+|            |               |    PEP     |
++------------+               +------------+
+      ^                            ^
+      |                            |
+      |                            |
+      v                            v
++------------+               +------------+
+|            |               |            |
+|  Identity  |               |    PDP     |
+|   Server   |               | (optional) |
+|            |               |            |
++------------+               +------------+
 ~~~
 
+The Identity Server provisions credentials to each of the workloads. At least Workload A (and possibly both) must be provisioned
+with a credential before the call can proceed. Details of communication with the Identity Server are out of scope
+of this document, however we do describe the credential received by the workload.
+
+PEP is a Policy Enforcement Point, the component that allows the call to go through or blocks it. PDP is an optional
```

Comment:
This is very opinionated about authorization in my opinion. I propose to mention authz but not include it into the high level architecture. 
What do others think?

Reply:
In what way is this opinionated in your opinion? :-) All I'm saying is that Service B is expected to make/enforce an authz decision. I'm just using common terminology from the policy world to do it.

Reply:
Let's discuss on Thursday. Resolving this for now to not block the PR.